### DISCUSSION TITLE
Questions about Svelte 5 Compatibility

### DISCUSSION THREAD

**Discussion by 0x5374656c6c6172** [melt-ui/melt-ui#957] [Q&A] [answered]
Hey,

1. I'm curious to know if there are any plans for Melt UI to transition to the Svelte 5 specification. If so, could you provide some details on the timeline or roadmap?

If the library is indeed moving to Svelte 5, I'd like to understand the potential impact on existing codebases:

2. Are there anticipated challenges or complexities in updating existing code that utilizes Melt UI when transitioning to Svelte 5?

Additionally, I'm currently working on a project that involves using Svelte 5:

3. Have there been any reported difficulties or challenges in integrating Melt UI with projects built on Svelte 5?

**Comment by arjendeblok** [comment]
I am also very curious about this. I just watched a video explaining how Melte does states and events the 'old way'. And I am curious if they also going to move to the new state and event patterns in Svelte 5.

**Comment by TGlide** [comment]
We're working on it 😄 https://github.com/melt-ui/runes-experiment

**Reply by karianpour** [reply]
meanwhile you might consider just making the peer dependencies fixed:
```
-  "svelte": "^3 || ^4"
+  "svelte": "^3 || ^4 || ^5.0.0-next.118"
```
Svelte 5 is backward compatible.

**Comment by arjendeblok** [comment]
@TGlide  Nice! I am just watching a video with you and CodingCatDev where you demonstrate the builder pattern with runes. Looks very promising.

**Comment by o-az** [comment]
I made this pr that updates the svelte peer dependency to allow svelte: https://github.com/melt-ui/melt-ui/pull/1207

https://github.com/melt-ui/melt-ui/pull/1207/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R61

I hope it gets accepted.

**Comment by TGlide** [comment, marked as answer]
Melt is now Svelte 5 compatible!

**Reply by vvvyyynet** [reply]
With `"svelte": "^5.0.0-next.143"` and `npm install -D @melt-ui/svelte` I still get the following errors. What do I need to change? Thank you!

```
sh-5.2$ npm install -D @melt-ui/svelte
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: 2024-05-testing-ui-libraries@0.0.1
npm ERR! Found: svelte@5.0.0-next.143
npm ERR! node_modules/svelte
npm ERR!   dev svelte@"^5.0.0-next.143" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer svelte@">=3 <5" from @melt-ui/svelte@0.79.1
npm ERR! node_modules/@melt-ui/svelte
npm ERR!   dev @melt-ui/svelte@"*" from the root project
```

**Reply by TGlide** [reply]
Sorry, forgot to release!

**Reply by vvvyyynet** [reply]
Thanks @TGlide. Will there soon be a release on the **main branch**, too, or shall I install the **dev branch** instead? For the latter case: Is there an npm command for this or do I need to do this manually?

**Reply by TGlide** [reply]
Branches are not associated with releases. It should already have been released in npm

**Reply by vvvyyynet** [reply]
You are right, npm has already updated. Unfortunately, I still get the same dependency error message, just now with 
`npm ERR! peer svelte@">=3 <=5" from @melt-ui/svelte@0.80.0` instead of 
`npm ERR! peer svelte@">=3 <5" from @melt-ui/svelte@0.79.1`.

**Reply by vvvyyynet** [reply]
Ok, I got everything to work with `npm install -D @melt-ui/svelte --force`. Still, it was very confusing. Can it be that the `-next.143` postfix is somehow not supported with `svelte@">=3 <=5"`?

**Reply by TGlide** [reply]
Yeah, it seems the range provided needs to be more specific. Will change it again

**Reply by TGlide** [reply]
@vvvyyynet updated the package, let me know how it goes for you!

**Reply by vvvyyynet** [reply]
It works :+1: I've been able to `npm install -D @melt-ui/svelte` version 0.81.0 with warnings but without errors. Thank you so much!
